code/src/com/sixsq/nuvla/server/resources/deployment_set.clj

```diff
@@ -192,20 +192,34 @@ These resources represent a deployment set that regroups deployments.
   (assoc resource :operational-status (divergence-map resource request)))
 
 (defn assoc-next-refresh
-  [resource]
-  (assoc resource :next-refresh (t/to-str (t/plus (t/now) (t/duration-unit 1 :minutes)))))
+  [{:keys [auto-update-interval] :as resource}]
+  (assoc resource :next-refresh
+                  (-> (t/now)
+                      (t/plus (t/duration-unit (or auto-update-interval 1) :minutes))
+                      t/to-str)))
+
+(defn assoc-auto-update-flag
+  [{:keys [auto-update next-refresh] :as resource}
+   {{:keys [uuid]} :params :as request}]
+  (let [current                  (when (and auto-update uuid)
```

Comment:
why do you need to retrieve current as admin since you receive the current document in the arguments?

Reply:
Will tackle as part of https://github.com/SixSq/tasklist/issues/3370